### DISCUSSION TITLE
Detection is working really well, is there a way to add a tracker to it?

### DISCUSSION THREAD

**Discussion by xoxolo** [ultralytics/yolov5#7382] [General]
Good day,
I'm trying this on video feed / different computer configuration.
My labels are good generally, but no tracking creates:
- overcounting elements.
- different results from one machine to another due to a different frame rate.
Any clue on a solution for this?
Thanks a lot!
O

**Comment by glenn-jocher** [comment]
@xoxolo 👋 Hello! Thanks for asking about object **tracking** in computer vision. YOLOv5 🚀 is an object **detector** that detect, localizes and classifies objects in a single image. It does not connect objects across multiple images, for this you need a tracking solution. A few possible tracking solutions are:

- **Extended Kalman Filter (EKF)**: https://en.wikipedia.org/wiki/Kalman_filter
- **KLT tracker**: https://en.wikipedia.org/wiki/Kanade%E2%80%93Lucas%E2%80%93Tomasi_feature_tracker
- **YOLOv5 DeepSort tracker**: https://github.com/mikel-brostrom/Yolov5_DeepSort_Pytorch

Good luck 🍀 and let us know if you have any other questions!
